aodndata/soop/soop_co2_classifier.py

```diff
@@ -0,0 +1,129 @@
+VESSEL_CODE = {'AA': 'VNAA',
+               'IN': 'VLMJ'}
+
+
+class SoopCo2FileClassifier(FileClassifier):
+
+    @classmethod
+    def def_project(cls, src_file):
+        """
+        Define project name according to file name or extension
+        Differentiate IMOS SOOP-CO2 RealTime from Delayed Mode and Future_Reef_MAP files
+        eg : (DM) IMOS_SOOP-CO2_GST_20121027T045200Z_VLHJ_FV01.nc
+             (DM) FutureReefMap_GST_20140530T185029Z_9V2768_FV01.nc
+             <Project_Name>_<<Facility-Code>>_<Data-Code>_<Start-date>_<Platform-Code>_FV<File-Version>.nc
+             (RT) IN_2017-165-0000dat.txt
+              <Vessel_code>_yyyy-ddd-hhmmdat.tx
+              :type src_file: object
+        """
+
+        if src_file.name.endswith('.nc'):
+            fields = cls._get_file_name_fields(src_file.name, min_fields=5)
+            if 'FV00' in src_file.name:
+                return fields[1] + '_RT'  # <Facility-Name + RTnc suffix>
+            else:
+                return fields[0]  # <Project-Name>
+        else:
+            raise InvalidFileFormatError(
+                "'{name}'is not a valid RT CO2 data (*dat.txt) or not a NetCDF file.".format(
+                    name=src_file.name)
+            )
+
+    @classmethod
+    def dest_path(cls, src_file):
+
+        dir_list = []
+        project = cls.def_project(src_file)
+        if project not in VALID_PROJECT:
+            raise InvalidFileNameError(
+                "Invalid project name '{project}'. "
+                "Project should be IMOS, SOOP-CO2_RT or Future_Reef_MAP".format(project=project))
+
+        if project in ['IMOS', 'SOOP-CO2_RT']:
+            fields = cls._get_file_name_fields(src_file.name)
+            ship_code = fields[4]
+            if ship_code not in ship_callsign_list():
+                raise InvalidFileNameError(
+                    "Missing vessel callsign in file name '{name}'.".format(name=src_file.name))
+
+            project_base = 'IMOS'
+            facility = fields[1][:4]
+            sub_facility = fields[1]
+            platform = "%s_%s" % (ship_code, ship_callsign(ship_code))
+            dir_list.extend([project_base, facility, sub_facility, platform])
+
+        if project == 'FutureReefMap':
+            fields = cls._get_file_name_fields(src_file.name, min_fields=5)
+            ship_code = fields[3]
+            if ship_code not in ship_callsign_list():
+                raise InvalidFileNameError(
+                    "Missing vessel callsign in file name '{name}'.".format(name=src_file.name))
+
+            dir_list.append('Future_Reef_MAP')
+            data_type = 'underway'
+            dir_list.extend([data_type, ship_callsign(ship_code)])
+
+        if project in ['IMOS', 'FutureReefMap']:
+            att_list = cls._get_nc_att(src_file.src_path, ['cruise_id', 'time_coverage_start'])
+            year = att_list[1][:4]
+            cruise_id = att_list[0]
+            if not cruise_id:
```

Comment:
You don't need to check for this. If the attribute is missing, the same exception is raised in the `_get_nc_att()` method.